src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -369,13 +369,19 @@ bool Creature::InitEntry(uint32 entry, CreatureData const* data /*= nullptr*/)
     SetByteValue(UNIT_FIELD_BYTES_0, 2, minfo->gender);
 
     // Load creature equipment
-    if (data && data->equipmentId != 0)
+    if (data)
     {
-        m_originalEquipmentId = data->equipmentId;
-        LoadEquipment(data->equipmentId);
+        if (data->equipmentId != 0)
+        {
+            // If creature exist in `creature` table we must use `creature.equipment_id` (can be -1, it means random equipment)
+            m_originalEquipmentId = data->equipmentId;
+            LoadEquipment(data->equipmentId);
+        }
+        else
+            LoadEquipment(0, true); // Line in `creature` table exist, but `creature.equipment_id` == 0. Creature must spawn without equipment.
```

Comment:
This is definitely not correct - the intent is that creature.equipment_id does not have to use a value different than 0 for a "default selection"
Want one spawn have a different equipment than template? use creature.equipment_id
Want one spawn without equipment? define no default in creature_template and set creature.equipment_id to all spawns except then one you want to exclude (rare use case)

Summing up, this change breaks all existing `creature` spawns that rely on the behavior I described (which was what was intended when this got implemented)